### PR TITLE
Coded script that uses Twitter API to pull data

### PR DESCRIPTION
**Pull request by puskini33**
* wrote script that uses Twython(Twitter API) to retrieve data from Twitter
* wrote scripts to access the access_key

### CODE REVIEW

File: src/data/data_retrieval/Twitter_stream_data_retrieval_engine.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    # received data
+    def on_success(self, data: dict) -> None:
+        if self.streaming_minutes is not None:
+            self.verify_streaming_time()
+        if self.number_wanted_tweets is not None:
+            self.verify_number_streamed_tweets()
+
+        tweet_data = process_tweet(data)
+        save_to_csv(tweet_data)
```
Comment (KasiaCieslik) [inline src/data/data_retrieval/Twitter_stream_data_retrieval_engine.py:30]: CSV file doesn't have header. We need to research how to add them

File: src/data/data_retrieval/Twitter_stream_data_retrieval_engine.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        self.statuses.filter(track=keyword, language='en', tweet_mode='extended')
+
+    # received data
+    def on_success(self, data: dict) -> None:
+        if self.streaming_minutes is not None:
+            self.verify_streaming_time()
+        if self.number_wanted_tweets is not None:
+            self.verify_number_streamed_tweets()
+
+        tweet_data = process_tweet(data)
```
[the remark this answers is not in the document]
Comment (puskini33) [reply src/data/data_retrieval/Twitter_stream_data_retrieval_engine.py:29]: Yes, more information is necessary. I looked at the structure of the tweet and it is complicated (dictionary in a dictionary in a  dictionary; and for a tweet it takes a lot of time to go through and save everything), also the number of characteristics is huge. I selected a few more characteristics of the tweet. Let me know if they are good and if they are enough.

File: src/data/data_retrieval/Twitter_stream_data_retrieval_engine.py (hunk 1 above)
Comment (puskini33) [reply src/data/data_retrieval/Twitter_stream_data_retrieval_engine.py:30]: I have added. good idea!

File: src/data/data_analysis/utils/raw_tweet_processing.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+
+def process_tweet(tweet: dict) -> dict:
+    """Function filters out unwanted data and saves the hashtag, user name, text, location in a tweet dictionary."""
+    processed_tweet = {}
+
+    if 'entities' in tweet and 'hashtags' in tweet['entities']:
```
[the remark this answers is not in the document]
Comment (puskini33) [reply src/data/data_analysis/utils/raw_tweet_processing.py:6]: Thanks for this shortcut. It did reduce the amount of code by a lot. Unfortunately, get() does not work with getting the value of a dictionary inside another dictionary.

**Comment by puskini33** [comment]
# Pull Request should be merged with master

**Pull request closed without merging**